swan-lake/integration-tutorials/sending-a-message-to-a-service.md

```diff
@@ -35,56 +35,82 @@ Follow the instructions given in this section to develop the service.
     $ bal new sending-a-message-to-a-service
     ```
 
-2. Introduce the source code in files with the `.bal` extension (e.g., the `main.bal` file). 
+2. Remove the generated content in the `main.bal` file and open the [Ballerina HTTP API Designer](https://wso2.com/ballerina/vscode/docs/design-the-services/http-api-designer) in VS Code.
 
-    Import the 
-    - `ballerina/http` module to develop the REST API and define the client that can be used to send requests to the backend service
-    - `ballerina/log` module to log some information for each client request
-
-    ```ballerina
-    import ballerina/http;
-    import ballerina/log;
-    ```
+    ![Open diagram view](/learn/images/tutorial_sending_a_message_to_a_service_open_diagram_view.gif)
 
 3. Define two [configurable variables](https://ballerina.io/learn/by-example/#configurability) for the port on which the listener should listen and the URL of the backend service.
 
+    ![Define configurable variables](/learn/images/tutorial_sending_a_message_to_a_service_define_configurable_variables.gif)
+
+    The corresponding source code that is generated is as follows.
+
     ```ballerina
     configurable int port = 8290;
     configurable string healthcareBackend = "http://localhost:9090/healthcare";
     ```
 
-4. Define an [`http:Client` client](https://ballerina.io/learn/by-example/#http-client) to send requests to the backend service.
+4. Generate the record corresponding to the payload from the backend service using the "Paste JSON as record" VS Code command by providing a sample of the expected JSON payload.
 
-    ```ballerina
-    final http:Client queryDoctorEP = check initializeHttpClient();
-
-    function initializeHttpClient() returns http:Client|error => new (healthcareBackend);
+    ```json
+    {
+        "name": "thomas collins",
+        "hospital": "grand oak community hospital",
+        "category": "surgery",
+        "availability": "9.00 a.m - 11.00 a.m",
+        "fee": 7000
+    }
     ```
 
-    The argument to the `new` expression is the URL for the backend service.
+    ![Paste JSON as record](/learn/images/tutorial_sending_a_message_to_a_service_paste_json_as_record.gif)
+    
+    1. Copy the JSON.
+    2. Open VS [Code Command palette](https://code.visualstudio.com/docs/getstarted/userinterface#_command-palette).
+    3. Search and select `Ballerina: Paste JSON as record` command.
+    
+    > **Note:** You can manually define the record, if there is no sample JSON.
+    > 
+    > ```ballerina
+    > type Doctor record {|
+    >     string name;
+    >     string hospital;
+    >     string category;
+    >     string availability;
+    >     decimal fee;
+    > |};
+    > ```
 
-5. Define a record corresponding to the payload from the backend service.
+    The payload will be an array of JSON objects in which, the structure of each JSON object matches this record.
```

Comment:
Since it says the payload will be an array of JSON objects, isn't it better to use an json array as the sample payload above which was used to paste?

Reply:
if so, we get something like this 
```
type NewRecordItem record {
    string name;
    string hospital;
    string category;
    string availability;
    int fee;
};

type NewRecord record {
    NewRecordItem[] newRecord;
};
```

But we don't want this `NewRecord`

Reply:
@MaryamZi / @hevayo Is this a usability improvement for editor?

Reply:
Yeah, I think adding the array as a field of another record is not necessary.

I think what's more appropriate here would be generating an array type (if at all).

```ballerina
type NewRecordItem record {
    string name;
    string hospital;
    string category;
    string availability;
    int fee;
};

type NewRecordArray NewRecordItem[];
```